app/controllers/ems_cloud_controller.rb

```diff
@@ -235,7 +266,23 @@ def set_ems_record_vars(ems, mode = nil)
 
     ems.azure_tenant_id = params[:azure_tenant_id] if ems.kind_of?(ManageIQ::Providers::Azure::CloudManager)
 
-    ems.update_authentication(build_credentials(ems), :save => (mode != :validate))
```

Comment:
@juliancheal Carefully take another look at this; this appears to be what's ailing you in that spec. Looks like this method, [from the authentication mixin](https://github.com/manageiq/manageiq/blob/5af68deb03d0f07c78e0c79ea6878ff27c648788/app/models/mixins/authentication_mixin.rb#L120), is what sets userid and password in master and not in this PR anymore.